lab_share_lib/processing/rabbit_message.py

```diff
@@ -15,10 +16,10 @@
 
 
 class RabbitMessage:
-    def __init__(self, headers, encoded_body):
+    def __init__(self, headers, encoded_body, reader_schema_version=None):
```

Comment:
Out of interest, why have a default for this?  I assume that missing the reader schema out of the config would be a critical error and that we shouldn't try to decode messages without it.

Reply:
Ah I see now I've read further down.  You need to have the RabbitMessage instance to be able to get the subject for the message.  Instead of trying to set the reader schema on the RabbitMessage instance then, can it be passed into the `decode` method call?  We don't need this schema at any other time AFAIK.

Reply:
My first impression (in fact, implementation) was to pass it to the `decode` call. But it felt weird having the reader schema from the properties and the writer schema from the method arguments. But yes, it can be done.

Reply:
I think it makes more sense as an argument.  In theory you could decode the message multiple ways with different reader schemas.  I know we don't, but it would be weird to have to change a property on the object to make a method come up with different answers.